osv2/bulk.go

```diff
@@ -0,0 +1,180 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+// BulkRequest is a serializable form of [opensearchtools.BulkRequest] specific to
+// the [opensearchapi.BulkRequest] in OpenSearch V2.
+// An empty BulkRequest will fail to execute. At least one Action is required to be added.
+//
+// For more details see https://opensearch.org/docs/latest/api-reference/document-apis/bulk/
+type BulkRequest struct {
+	// Actions lists the actions to be performed in the BulkRequest
+	Actions []opensearchtools.BulkAction
+
+	// Refresh determines if the request should wait for a refresh or not
+	Refresh opensearchtools.Refresh
+
+	// Index determines the entire index for the request
+	Index string
+}
+
+// fromDomainBulkRequest creates a new [BulkRequest] from the given [opensearchtools.BulkRequest/.
+func fromDomainBulkRequest(req *opensearchtools.BulkRequest) (BulkRequest, opensearchtools.ValidationResults) {
+	// As more versions are implemented, these [opensearchtools.ValidationResults] will be used to contain issues
+	// converting from the domain model to the V2 model.
+	var vrs opensearchtools.ValidationResults
+
+	return BulkRequest{
+		Actions: req.Actions,
+		Refresh: req.Refresh,
+		Index:   req.Index,
+	}, vrs
+}
+
+// Validate validates the given BulkRequest
+func (r *BulkRequest) Validate() opensearchtools.ValidationResults {
```

Comment:
Does this need to be exported?

Reply:
I think I would export them.  Validation is a reasonable part of a "domain" API IMO.